cpp/include/graph.hpp

```diff
@@ -214,5 +214,182 @@ class GraphCSC: public GraphCompressedSparseBase<VT,ET,WT> {
+   *
+   * @param  offsets               This array of size V+1 (V is number of vertices) contains the offset of adjacency lists of every vertex.
+   *                               Offsets must be in the range [0, E] (number of edges).
+   * @param  indices               This array of size E contains the index of the destination for each edge.
+   *                               Indices must be in the range [0, V-1].
+   * @param  edge_data             This array of size E (number of edges) contains the weight for each edge.  This
+   *                               array can be null in which case the graph is considered unweighted.
+   * @param  number_of_vertices    The number of vertices in the graph
+   * @param  number_of_edges       The number of edges in the graph
+   */
+  ConstructedGraphCSR(std::unique_ptr<ET> &&offsets_,
+                      std::unique_ptr<VT> &&indices_,
+                      std::unique_ptr<WT> &&edge_data_,
+                      VT number_of_vertices_,
+                      ET number_of_edges_):
+    ConstructedGraphCompressedSparseBase<VT,ET,WT>(offsets_, indices_, edge_data_, number_of_vertices_, number_of_edges_)
+  {}
+};
+
+/**
+ * @brief       A constructed graph stored in CSC (Compressed Sparse Column) format.
+ *
+ * @tparam VT   Type of vertex id
+ * @tparam ET   Type of edge id
+ * @tparam WT   Type of weight
+ */
+template <typename VT, typename ET, typename WT>
+class ConstructedGraphCSC: public ConstructedGraphCompressedSparseBase<VT,ET,WT> {
+public:
+  /**
+   * @brief      Default constructor
+   */
+  ConstructedGraphCSC(): ConstructedGraphCompressedSparseBase<VT,ET,WT>() {}
+  
+  /**
+   * @brief      Take ownership of the provided graph arrays in CSR format
+   *
+   * @param  offsets               This array of size V+1 (V is number of vertices) contains the offset of adjacency lists of every vertex.
+   *                               Offsets must be in the range [0, E] (number of edges).
+   * @param  indices               This array of size E contains the index of the destination for each edge.
+   *                               Indices must be in the range [0, V-1].
+   * @param  edge_data             This array of size E (number of edges) contains the weight for each edge.  This array
+   *                               can be null in which case the graph is considered unweighted.
+   * @param  number_of_vertices    The number of vertices in the graph
+   * @param  number_of_edges       The number of edges in the graph
+   */
+  ConstructedGraphCSC(std::unique_ptr<ET> &&offsets_,
+                      std::unique_ptr<VT> &&indices_,
+                      std::unique_ptr<WT> &&edge_data_,
+                      VT number_of_vertices_,
+                      ET number_of_edges_):
+    ConstructedGraphCompressedSparseBase<VT,ET,WT>(offsets_, indices_, edge_data_, number_of_vertices_, number_of_edges_)
+  {}
+};
+
```

Comment:
And why are we creating this new class? Can we just return existing graph class objects?

Reply:
Probably worth following up in #799